### PR TITLE
GeoServer REST importer

### PR DESCRIPTION
**Pull request by dnlkoch**
This adds the `GeoServerRESTImporter` that can be used to easily call the GeoServer importer extension's REST API.

### CODE REVIEW

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java [hunk 1]
```diff
@@ -0,0 +1,701 @@
...
+	 *
+	 */
+	public GeoServerRESTImporter() {
+
+	}
+
+	/***
+	 *
+	 * @param importerBaseURL
+	 * @param defaultSRS
```
Comment (ahennr) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java]: not needed

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+		this.username = username;
+		this.password = password;
+
+		this.mapper = mapper;
+		this.baseUri = new URI(importerBaseURL);
+	}
+
+	/**
+	 *
+	 * @param importJob
```
Comment (ahennr) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java]: wrong javadoc parameters

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataDirectory.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import de.terrestris.shogun2.importer.communication.RESTData;
+
+/**
+ *
+ * @author Daniel Koch
+ * @author terrestris GmbH & Co. KG
+ *
+ */
+public class RESTDataDirectory extends RESTData {
```
Comment (ahennr) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataDirectory.java]: Is this class needed currently?

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataFile.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+import de.terrestris.shogun2.importer.communication.RESTData;
+
+/**
+ *
+ * @author Daniel Koch
+ * @author terrestris GmbH & Co. KG
+ *
+ */
+public class RESTDataFile extends RESTData {
```
Comment (ahennr) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataFile.java]: Is this class needed currently?

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataRemote.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+import de.terrestris.shogun2.importer.communication.RESTData;
+
+/**
+ *
+ * @author Daniel Koch
+ * @author terrestris GmbH & Co. KG
+ *
+ */
+public class RESTDataRemote extends RESTData {
```
Comment (ahennr) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataRemote.java]: Is this class needed currently?

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/transform/RESTDateFormatTransform.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+import de.terrestris.shogun2.importer.transform.RESTTransform;
+
+/**
+ * 
+ * @author Daniel Koch
+ * @author terrestris GmbH & Co. KG
+ *
+ */
+public class RESTDateFormatTransform extends RESTTransform {
```
Comment (ahennr) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/transform/RESTDateFormatTransform.java]: Is this class needed currently?

File: src/pom.xml [hunk 7]
```diff
@@ -193,6 +197,12 @@
         <!-- Apache HTTP Client -->
         <apache-httpclient.version>4.5.3</apache-httpclient.version>
 
+        <!-- GeoTools -->
+        <geotools.version>17.3</geotools.version>
```
Comment (ahennr) [inline src/pom.xml]: Does this version depend on the used version of GeoServer resp. the used version of the importer?

Comment (dnlkoch) [reply src/pom.xml]: No, actually not.

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataDirectory.java (hunk 3 above)
Comment (dnlkoch) [reply src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataDirectory.java]: No, actually not. I would leave it for future purposes, ok?

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataFile.java (hunk 4 above)
Comment (dnlkoch) [reply src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataFile.java]: See above.

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataRemote.java (hunk 5 above)
Comment (dnlkoch) [reply src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataRemote.java]: See above.

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/transform/RESTDateFormatTransform.java (hunk 6 above)
Comment (dnlkoch) [reply src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/transform/RESTDateFormatTransform.java]: See above.

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataDirectory.java (hunk 3 above)
Comment (ahennr) [reply src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/communication/RESTDataDirectory.java]: Alright.

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java (hunk 1 above)
Comment (dnlkoch) [reply src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java]: Fixed in b119a96e9828b8def01eddd0fd0dd3ede1fe207a.

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java (hunk 2 above)
Comment (dnlkoch) [reply src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java]: Fixed in b119a96e9828b8def01eddd0fd0dd3ede1fe207a.

File: src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java [hunk 8]
```diff
@@ -0,0 +1,701 @@
...
+
+	/**
+	 *
+	 */
+	private ObjectMapper mapper = new ObjectMapper();
+
+	/**
+	 *
+	 */
+	private URI baseUri;
```
Comment (KaiVolland) [inline src/shogun2-core/src/main/java/de/terrestris/shogun2/importer/GeoServerRESTImporter.java]: PLz provide some more docs.

**Comment by ahennr** [comment]
Will merge this PR now and create an issue for enhancement of the docs

**Pull request merged**